packages/babel-parser/src/options.ts

```diff
@@ -104,5 +104,9 @@ export function getOptions(opts?: Options | null): OptionsWithDefaults {
   for (const key of Object.keys(defaultOptions) as (keyof Options)[]) {
     options[key] = opts[key] ?? defaultOptions[key];
   }
-  return options;
```

Comment:
Here we starts from `{}` and changed the shape for ~20 times. Can we start the new options from `{ ...defaultOptions }` and assign the values if `opts[key]` is truish? Doing so should also avoid the shape migration during initialization.